spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/integration/outbound/PubSubOutboundChannelAdapter.java

```diff
@@ -0,0 +1,32 @@
+package org.springframework.cloud.gcp.pubsub.integration.outbound;
+
+import com.google.cloud.pubsub.spi.v1.Publisher;
+import com.google.pubsub.v1.PubsubMessage;
+import com.google.pubsub.v1.TopicName;
+import java.io.IOException;
+import org.springframework.integration.handler.AbstractMessageHandler;
+import org.springframework.messaging.Message;
+
+/**
+ * Created by joaomartins on 5/19/17.
+ */
+public class PubSubOutboundChannelAdapter extends AbstractMessageHandler {
+
+  private String projectId;
+
+  private Publisher publisher;
+
+  public PubSubOutboundChannelAdapter(String projectId, String topicName) throws IOException {
+    publisher = Publisher.defaultBuilder(TopicName.create(
+        projectId, topicName)).build();
+  }
+
+  @Override
+  protected void handleMessageInternal(Message<?> message) throws Exception {
+    if (!(message.getPayload() instanceof PubsubMessage)) {
+      throw new IllegalArgumentException("This channel adapter cannot send messages other than"
```

Comment:
That's fully dead end. If it can't it must create one.
That's why in the Desing Doc I stand out against `PubSubMessagingTemplate` idea. It doesn't provide expected flexibility and makes end-users to confuse.

See `KinesisMessageHandler` for example how the target entity can be built: https://github.com/spring-projects/spring-integration-aws/blob/master/src/main/java/org/springframework/integration/aws/outbound/KinesisMessageHandler.java